storage/xtradb/handler/handler0alter.cc

```diff
@@ -3013,6 +3148,23 @@ prepare_inplace_alter_table_dict(
 			ctx->new_table->fts->doc_col = fts_doc_id_col;
 		}
 
+		temp_def = index_defs;
+		for (ulint i = 0; i < ctx->num_to_add_index; i++, temp_def++) {	
+			if (temp_def->ind_type & DICT_UNIQUE_HASH) {
+				res = static_cast<char*>(mem_heap_alloc(ctx->heap, strlen("HASH_COL_") + 
+				strlen(temp_def->name) +1));
+				memcpy(res,"HASH_COL_", strlen("HASH_COL_"));
```

Comment:
I have seen this before, can you create a function that does this and use this function on every place?